src/routes/safe/components/Apps/trackAppUsageCount.ts

```diff
@@ -0,0 +1,61 @@
+import { SafeApp } from 'src/routes/safe/components/Apps/types'
+import local from 'src/utils/storage/local'
+
+export const APPS_DASHBOARD = 'APPS_DASHBOARD'
+
+const TX_COUNT_WEIGHT = 2
+const OPEN_COUNT_WEIGHT = 1
+const MORE_RECENT_MULTIPLIER = 2
+const LESS_RECENT_MULTIPLIER = 1
+
+export type AppTrackData = {
+  [safeAppId: string]: {
+    timestamp: number
+    openCount: number
+    txCount: number
+  }
+}
+
+export const countOpen = (app: SafeApp): void => {
+  const trackData = local.getItem<AppTrackData>(APPS_DASHBOARD) || {}
+  const currentOpenCount = trackData[app.id]?.openCount || 0
+  const currentTxCount = trackData[app.id]?.txCount || 0
+  local.setItem(APPS_DASHBOARD, {
+    ...trackData,
+    [app.id]: {
+      ...trackData[app.id],
+      timestamp: Date.now(),
+      openCount: currentOpenCount + 1,
+      txCount: currentTxCount,
+    },
+  })
+}
+
+export const countTxs = (app: SafeApp): void => {
+  const trackData = local.getItem<AppTrackData>(APPS_DASHBOARD) || {}
+  const currentTxCount = trackData[app.id]?.txCount || 0
+  local.setItem(APPS_DASHBOARD, {
+    ...trackData,
+    [app.id]: { ...trackData[app.id], txCount: currentTxCount + 1 },
+  })
+}
+
+export const rankTrackedSafeApps = (apps: AppTrackData): string[] => {
+  const appsMap = Object.entries(apps)
+
+  return appsMap
+    .sort((a, b) => {
+      // The more recently used app gets a bigger score/relevancy multiplier
+      const aTimeMultiplier =
+        a[1].timestamp.valueOf() - b[1].timestamp.valueOf() > 0 ? MORE_RECENT_MULTIPLIER : LESS_RECENT_MULTIPLIER
```

Comment:
Aren't the timestamps already a primitive?

Reply:
Oh nice catch! Was a leftover as I had them as type `Date` before